components/script/webdriver_handlers.rs

```diff
@@ -91,6 +91,40 @@ fn is_stale(element: &Element) -> bool {
     !element.owner_document().is_active() || !element.is_connected()
 }
 
+/// <https://w3c.github.io/webdriver/#dfn-disabled>
+fn is_disabled(element: &Element) -> bool {
+    // Step 1. If element is an option element or element is an optgroup element
+    if element.is::<HTMLOptionElement>() || element.is::<HTMLOptGroupElement>() {
+        // Step 1.1. For each inclusive ancestor ancestor of element
+        let disabled = element
+            .upcast::<Node>()
+            .inclusive_ancestors(ShadowIncluding::No)
+            .any(|node| {
+                if node.is::<HTMLOptGroupElement>() || node.is::<HTMLSelectElement>() {
+                    // Step 1.1.1. If ancestor is an optgroup element or ancestor is a select element,
+                    // and ancestor is actually disabled, return true.
+                    if let Some(element) = node.downcast::<Element>() {
+                        element.is_actually_disabled()
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                }
+            });
+
+        // Step 1.2
+        // The spec suggests that we immediately return false if the above is not true.
+        // However, it causes disabled option element to not be considered as disabled.
+        // Hence, here we also check if the element itself is actually disabled.
+        if disabled {
+            return true;
+        }
```

Comment:
```suggestion
        return disabled;
```

If it is option element, then we won't enter step 1 in the first place. It exposes other problems. Just follow the spec.

Reply:
actually steps 1 include option element https://w3c.github.io/webdriver/#dfn-disabled

> 1. If element is an [option](https://html.spec.whatwg.org/multipage/form-elements.html#the-option-element) element or element is an [optgroup](https://html.spec.whatwg.org/multipage/form-elements.html#the-optgroup-element) element:
> 
> 1.1. For each [inclusive ancestor](https://dom.spec.whatwg.org/#concept-tree-inclusive-ancestor) ancestor of element:
> 
> 1.1.1. If ancestor is an [optgroup](https://html.spec.whatwg.org/multipage/form-elements.html#the-optgroup-element) element or ancestor is a [select](https://html.spec.whatwg.org/multipage/form-elements.html#the-select-element) element, and ancestor is [actually disabled](https://w3c.github.io/webdriver/#dfn-actually-disabled), return true.
> 
> 1.2. Return false.

You can see that if it's an option it will go to steps 1.1, but in steps 1.1.1. it will only check optgroup and select.

Reply:
So you mean the spec has some problem here?

Reply:
I think yes